build-system/tasks/release/index.js

```diff
@@ -0,0 +1,370 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const experimentsConfig = require('../../global-configs/experiments-config.json');
+const fetch = require('node-fetch');
+const fs = require('fs-extra');
+const log = require('fancy-log');
+const path = require('path');
+const tar = require('tar');
+const {cyan, green, yellow} = require('ansi-colors');
+const {execOrDie} = require('../../common/exec');
+const {MINIFIED_TARGETS} = require('../helpers');
+const {VERSION} = require('../../compile/internal-version');
+
+// Flavor config for the base flavor type.
+const BASE_FLAVOR_CONFIG = {
+  flavorType: 'base',
+  name: 'base',
+  rtvPrefixes: ['00', '01', '02', '03', '04', '05'],
+  // TODO(#28168, erwinmombay): relace with single `--module --nomodule` command.
+  command: 'gulp dist --noconfig --esm && gulp dist --noconfig',
+  environment: 'AMP',
+};
+
+// Deep map from [environment][flavorType] to RTV prefix.
+const EXPERIMENTAL_RTV_PREFIXES = {
+  AMP: {
+    experimentA: '10',
+    experimentB: '11',
+    experimentC: '12',
+  },
+  INABOX: {
+    experimentA: '20',
+    experimentB: '22',
+    experimentC: '24',
+  },
+};
+
+// Directory containing multiple static files to dist/.
+const STATIC_FILES_DIR = path.resolve(__dirname, 'static');
+
+// List of individual files to copy directly from the Git workspace to dist/.
+const STATIC_FILE_PATHS = ['build-system/global-configs/caches.json'];
+
+// List of individual files to copy from post-build files, key-values as the
+// from-to of the copy operations.
+const POST_BUILD_MOVES = {
+  'dist.tools/experiments/experiments.cdn.html': 'dist/experiments.html',
+  'dist.tools/experiments/experiments.js': 'dist/v0/experiments.js',
+  'dist.tools/experiments/experiments.js.map': 'dist/v0/experiments.js.map',
+};
+
+// URL to the JSON info API for the amp-sw package on the npm registry.
+const AMP_SW_NPM_PACKAGE_URL = 'https://registry.npmjs.org/@ampproject/amp-sw';
+
+// List of directories to keep in the temp directory from each flavor's build.
+const DIST_DIRS = ['dist', 'dist.3p', 'dist.tools'];
+
+// Mapping from a channel's RTV prefix, to an object with channel configuration
+// data based on the spec/amp-framework-hosting.md document. The fields are:
+// - type: machine name of the channel (Some of these are the same as their name
+//   in spec/amp-framework-hosting.md, but some are different. Those that have a
+//   different machine name are marked in a comment.)
+// - configBase: name of the config JSON file to prepend as the AMP_CONFIG to
+//   all the entry files in this channel. Either `canary` or `prod` (these files
+//   are located in build-system/global-configs/${configBase}-config.json).
+const CHANNEL_CONFIGS = {
+  '00': {type: 'experimental', configBase: 'canary'},
+  '01': {type: 'production', configBase: 'prod'}, // Spec name: 'stable'
```

Comment:
What would it take to rename the `production` and `rc` types to `stable` and `beta`? I ask because we were able to rename the `canary` type to `experimental`, so I'd imagine it is technically feasible.

Obviously, not suggesting it be done in this PR. But I'm hoping that moving all our release code to open source will make changes like this considerably easier.

(You're probably tired of my asking this question again and again 😃)

Reply:
It'll take getting @jridgewell on board with the idea

Reply:
I see. Let's punt this discussion, since it shouldn't block this PR.